addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -198,11 +198,37 @@ def migrate_properties(cr):
 
 
 def remove_account_moves_from_special_periods(cr):
-    """We first search for journal entries in a special period, in the
+    """ We first add the balance of lines on the centralization account of each
+    opening entry and entity. Those balance lines use a temporary year earnings
+    account that will be converted to the standard 9999 account in post-migration.
+    Then we search for journal entries in a special period, in the
     first reported fiscal year of the company, and we take them out of the
     special period, into a normal period, because we assume that this is
     the starting balance of the company, and should be maintained.
     Then we delete all the moves associated to special periods."""
+
+    # Get all the move lines regarding year opening balance
+    cr.execute("""
+        SELECT aml.id
+        FROM account_journal aj
+          JOIN account_account aa ON aa.id = aj.default_debit_account_id
+          JOIN account_move_line aml ON aml.account_id = aj.default_debit_account_id
+          JOIN account_move am ON am.id = aml.move_id
+          JOIN account_period ap ON ap.id = aml.period_id
+        WHERE aj.type = 'situation' AND ap.special = true AND aml.centralisation IS NOT NULL
+          AND (aml.debit <> 0.0 OR aml.credit <> 0.0)
```

Comment:
Better add a clause `AND am.journal_id = aj.id;` because I see a handful of non-compliant moves in the special opening period *not* in the situation journal. Those need to be dealt with but not here.